packages/libs/eda/src/lib/map/analysis/mapTypes/plugins/BarMarkerMapType.tsx

```diff
@@ -0,0 +1,668 @@
+import {
+  MapTypeConfigurationMenu,
+  MarkerConfigurationOption,
+} from '../../MarkerConfiguration/MapTypeConfigurationMenu';
+import { BarPlotMarkerIcon } from '../../MarkerConfiguration/icons';
+import { TabbedDisplayProps } from '@veupathdb/coreui/lib/components/grids/TabbedDisplay';
+import MapVizManagement from '../../MapVizManagement';
+import Spinner from '@veupathdb/components/lib/components/Spinner';
+import { MapFloatingErrorDiv } from '../../MapFloatingErrorDiv';
+import { MapTypeHeaderCounts } from '../MapTypeHeaderCounts';
+import { ChartMarkerPropsWithCounts } from '../../hooks/standaloneMapMarkers';
+
+const displayName = 'Bar plots';
+
+export const plugin: MapTypePlugin = {
+  displayName,
+  ConfigPanelComponent,
+  MapLayerComponent,
+  MapOverlayComponent,
+  MapTypeHeaderDetails,
+};
+
+function ConfigPanelComponent(props: MapTypeConfigPanelProps) {
+  const {
+    apps,
+    analysisState,
+    appState,
+    geoConfigs,
+    updateConfiguration,
+    studyId,
+    studyEntities,
+    filters,
+  } = props;
+
+  const geoConfig = geoConfigs[0];
+  const subsettingClient = useSubsettingClient();
+  const configuration = props.configuration as BarPlotMarkerConfiguration;
+  const {
+    selectedVariable,
+    selectedValues,
+    binningMethod,
+    dependentAxisLogScale,
+  } = configuration;
+
+  const { entity: overlayEntity, variable: overlayVariable } =
+    findEntityAndVariable(studyEntities, selectedVariable) ?? {};
+
+  if (
+    overlayEntity == null ||
+    overlayVariable == null ||
+    !Variable.is(overlayVariable)
+  ) {
+    throw new Error(
+      'Could not find overlay variable: ' + JSON.stringify(selectedVariable)
+    );
+  }
+
+  const filtersIncludingViewport = useMemo(() => {
+    const viewportFilters = appState.boundsZoomLevel
+      ? filtersFromBoundingBox(
+          appState.boundsZoomLevel.bounds,
+          {
+            variableId: geoConfig.latitudeVariableId,
+            entityId: geoConfig.entity.id,
+          },
+          {
+            variableId: geoConfig.longitudeVariableId,
+            entityId: geoConfig.entity.id,
+          }
+        )
+      : [];
+    return [...(filters ?? []), ...viewportFilters];
+  }, [
+    appState.boundsZoomLevel,
+    geoConfig.entity.id,
+    geoConfig.latitudeVariableId,
+    geoConfig.longitudeVariableId,
+    filters,
+  ]);
+
+  const allFilteredCategoricalValues = useCategoricalValues({
+    overlayEntity,
+    studyId,
+    overlayVariable,
+    filters,
+  });
+
+  const allVisibleCategoricalValues = useCategoricalValues({
+    overlayEntity,
+    studyId,
+    overlayVariable,
+    filters: filtersIncludingViewport,
+    enabled: configuration.selectedCountsOption === 'visible',
+  });
+
+  const previewMarkerData = useMarkerData({
+    studyId,
+    filters,
+    studyEntities,
+    geoConfigs,
+    boundsZoomLevel: appState.boundsZoomLevel,
+    selectedVariable: configuration.selectedVariable,
```

Comment:
We already deconstructed these from `configuration` at line 89 so we could save a few characters?

Reply:
Yep!